File: badri-apparel-backend/src/main/java/com/badribhaiapparel/service/CouponService.java
package com.badribhaiapparel.service;

import com.badribhaiapparel.dto.CouponValidationResult;
import com.badribhaiapparel.entity.Coupon;
import com.badribhaiapparel.entity.User;
import java.util.List;

public interface CouponService {
    CouponValidationResult validateCoupon(String code, User user, Double cartTotal);
    List<Coupon> getAllCoupons();
    Coupon createCoupon(Coupon coupon);
    Coupon updateCoupon(Long id, Coupon coupon);
    void deleteCoupon(Long id);
    void incrementUsage(String code);
}

File: src/services/couponService.js
import axios from './axios';

const couponService = {
  validateCoupon: async (code, cartTotal) => {
    try {
      const response = await axios.post('/coupons/validate', { code, cartTotal });
      return response.data;
    } catch (error) {
      return error.response?.data || { success: false, message: 'Failed to validate coupon' };
    }
  },

  getAllCoupons: async () => {
    const response = await axios.get('/coupons');
    return response.data;
  },

  createCoupon: async (couponData) => {
    const response = await axios.post('/coupons', couponData);
    return response.data;
  },

  updateCoupon: async (id, couponData) => {
    const response = await axios.put(`/coupons/${id}`, couponData);
    return response.data;
  },

  deleteCoupon: async (id) => {
    const response = await axios.delete(`/coupons/${id}`);
    return response.data;
  }
};

export default couponService;
